cmd/cmdoptions/client.go

```diff
@@ -64,6 +83,11 @@ func (c *ClientOptions) Dial(metrics *Metrics, logger *zap.SugaredLogger) (clien
 	clientOptions.ConnectionOptions.TLS = tlsCfg
 	clientOptions.Logger = NewZapAdapter(logger.Desugar())
 	clientOptions.MetricsHandler = metrics.NewHandler()
+	authHeader := os.Getenv(AUTH_HEADER_ENV_VAR)
+	if c.AuthHeader != "" {
+		authHeader = c.AuthHeader // CLI argument value overrides env var value
+	}
+	clientOptions.HeadersProvider = newAuthHeaderProvider(authHeader)
```

Comment:
We should only set a header provider if there is an auth header.

Granted we'll probably be coming back and changing this anyways if/when we ever want any other header than just this one special one. Arguably you should just have a `headerProvider` that has an entire map of headers to set instead of assuming this is the only use of header provider we'll ever need.